doc/compe.txt

```diff
@@ -212,91 +371,49 @@ LUA CONFIG                                                  *compe-usage-ex-lua*
       vsnip = true;
       nvim_lsp = true;
       nvim_lua = true;
+      spell = true;
+      tags = true;
       snippets_nvim = true;
     };
   }
 >
 
-
-
 ==============================================================================
-MAPPINGS                                                  *compe-usage-mappings*
+CUSTOM SOURCE                                            *compe-custom-source*
 
-If you don't use any autopair plugin.
->
-  inoremap <silent><expr> <C-Space> compe#complete()
-  inoremap <silent><expr> <CR>      compe#confirm('<CR>')
-  inoremap <silent><expr> <C-e>     compe#close('<C-e>')
-<
+Source is defined as dictionary with items:
 
-If you use cohama/lexima.vim
->
-  inoremap <silent><expr> <C-Space> compe#complete()
-  inoremap <silent><expr> <CR>      compe#confirm(lexima#expand('<LT>CR>', 'i'))
-  inoremap <silent><expr> <C-e>     compe#close('<C-e>')
-<
-
-
-==============================================================================
-SOURCE CONFIGURATION                                        *compe-usage-source*
+get_metadata ~
+    This function should return the default source configuration.
+    See |compe-source| section.
 
-The sources can be configured by
+determine ~
+    This function should return dict as:
 >
-  let g:compe.source['source_name'] = { ...configuration... }
+    { keyword_pattern_offset   = 1-origin number;
+      trigger_character_offset = 1-origin number; }
 <
-in Vimscript or passing the configuration inside
+    If this function returns empty, nvim-compe will do nothing.
+
+complete ~
+    This function should callback the completed items as:
 >
-  sources['source_name']
+    args.callback({ items = items })
 <
-field in Lua setup function.
-
-- priority
-  Specify source priority.
-
-- filetypes
-  Specify source filetypes.
-
-- ignored_filetypes
-  Specify filetypes that should not use this source.
-
-- sort
-  Specify source is sortable or not.
-
-- dup
-  Specify source candidates can have the same word another item.
+    To stop the competion process, call `args.abort()`.
 
-- menu
-  Specify item's menu (see `:help complete-items`)
+documentation ~
+    Optional.
+    You can provide documentation for selected items.
+    See nvim_lsp or snippets.nvim source for examples.
 
+confirm ~
```

Comment:
Please add `resolve` for it.

Reply:
What `resolve` does? It wasn't in documentation before

Reply:
Looked up the LSP specs and still not entirely sure, but can it be something like this? At least for now
```
resolve
    Optional.
    Resolve completion item.
```

Reply:
Sorry. I had only added it to the wiki page.
https://github.com/hrsh7th/nvim-compe/wiki#resolve-optional

Reply:
Thank you!